### PR TITLE
CI: Use GitHub Container Registry for CI images

### PR DESCRIPTION
**Pull request by feelepxyz**
Attempting to use ghrc.io for CI images. This will probably not work on
forks.

### CODE REVIEW

**Comment by jurre** [comment]
Do we assume this will be faster, or are we trying to find out?

**Comment by feelepxyz** [comment]
@jurre yeah mostly trying, assuming it would be a lot faster to pull if also hosted on azure next to actions runners somewhere 🤞

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -50,14 +50,14 @@ jobs:
...
       - name: Log in to Docker registry
         run: |
-          if [ -n "${{ secrets.DOCKER_USERNAME }}" ] && [ -n "${{ secrets.DOCKER_PASSWORD }}" ]; then
-            echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
-            echo "DOCKER_LOGGED_IN=true" >> $GITHUB_ENV
-          else
+          if [[ -z "${{ secrets.GHCR_PAT }}" ]]; then
             echo "No Docker credentials, skipping login"
+          else
+            echo "${{ secrets.GHCR_PAT }}" | docker login ghcr.io -u ${{ github.actor }} --password-stdin
```
Comment (thepwagner) [inline .github/workflows/ci.yml:59]: Question: could this be `secrets.GITHUB_TOKEN`? https://github.blog/changelog/2021-03-24-packages-container-registry-now-supports-github_token/

It looks like you may have started there and fell back to a PAT.

Comment (thepwagner) [reply .github/workflows/ci.yml:59]: Question: who does this PAT belong to? It may be worth documenting here so access can be managed.

**Review by thepwagner — APPROVED** [review]
Noice! LGTM, though I think we should avoid the PAT if possible.

**Comment by feelepxyz** [comment]
Tried this here https://github.com/dependabot/dependabot-core/pull/3428 and didn't see any improvements, if anything the downloads where slower (around ~2mins vs 1.5min) and not sure it's storing inline caching as builds where often re-building from scratch even though there should have built from an up-to-date cache.

**Comment by feelepxyz** [comment]
Looks like buildkit support is on the radar but not done yet https://github.slack.com/archives/CJL5ATGUC/p1607096013102400

**Pull request closed without merging**